src/pages/workspace/members/WorkspaceOwnerPaymentCardFormPage.tsx

```diff
@@ -0,0 +1,293 @@
+
+function WorkspaceOwnerPaymentCardFormPage({route}: WorkspaceOwnerPaymentCardFormPageProps) {
+    const styles = useThemeStyles();
+    const theme = useTheme();
+    const {translate} = useLocalize();
+
+    const cardNumberRef = useRef<AnimatedTextInputRef>(null);
+
+    const [isCurrencyModalVisible, setIsCurrencyModalVisible] = useState(false);
+    const [currency, setCurrency] = useState<keyof typeof CONST.CURRENCY>(CONST.CURRENCY.USD);
+
+    const policyID = route.params.policyID;
+    const accountID = route.params.accountID;
+
+    useEffect(
+        () => {
+            PaymentMethods.clearDebitCardFormErrorAndSubmit();
+
+            return () => {
+                PaymentMethods.clearDebitCardFormErrorAndSubmit();
+            };
+        },
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+        [],
+    );
+
+    const validate = (values: FormOnyxValues<typeof ONYXKEYS.FORMS.ADD_DEBIT_CARD_FORM>): FormInputErrors<typeof ONYXKEYS.FORMS.ADD_DEBIT_CARD_FORM> => {
+        const errors = ValidationUtils.getFieldRequiredErrors(values, REQUIRED_FIELDS);
+
+        if (values.nameOnCard && !ValidationUtils.isValidLegalName(values.nameOnCard)) {
+            errors.nameOnCard = 'addDebitCardPage.error.invalidName';
+        }
+
+        if (values.cardNumber && !ValidationUtils.isValidDebitCard(values.cardNumber.replace(/ /g, ''))) {
+            errors.cardNumber = 'addDebitCardPage.error.debitCardNumber';
+        }
+
+        if (values.expirationDate && !ValidationUtils.isValidExpirationDate(values.expirationDate)) {
+            errors.expirationDate = 'addDebitCardPage.error.expirationDate';
+        }
+
+        if (values.securityCode && !ValidationUtils.isValidSecurityCode(values.securityCode)) {
+            errors.securityCode = 'addDebitCardPage.error.securityCode';
+        }
+
+        if (values.addressStreet && !ValidationUtils.isValidAddress(values.addressStreet)) {
+            errors.addressStreet = 'addDebitCardPage.error.addressStreet';
+        }
+
+        if (values.addressZipCode && !ValidationUtils.isValidZipCode(values.addressZipCode)) {
+            errors.addressZipCode = 'addDebitCardPage.error.addressZipCode';
+        }
+
+        return errors;
+    };
+
+    const addPaymentCard = useCallback(
+        (values: FormOnyxValues<typeof ONYXKEYS.FORMS.ADD_DEBIT_CARD_FORM>) => {
+            const cardData = {
+                cardNumber: values.cardNumber,
+                cardMonth: CardUtils.getMonthFromExpirationDateString(values.expirationDate),
+                cardYear: CardUtils.getYearFromExpirationDateString(values.expirationDate),
+                cardCVV: values.securityCode,
+                addressName: values.nameOnCard,
+                addressZip: values.addressZipCode,
+                currency: CONST.CURRENCY.USD,
+            };
+            Policy.addBillingCardAndRequestPolicyOwnerChange(policyID, cardData);
+            Navigation.dismissModal();
```

Comment:
We shouldn't dismiss the modal. Instead we should let `Policy.addBillingCardAndRequestPolicyOwnerChange` decide if we should dismiss the modal , i.e. no `changeOwner` errors or navigate the user to `WorkspaceOnerChangeCheckPage`

Reply:
same answer as before